front_end/Profile/UserProfile.js

```diff
@@ -0,0 +1,298 @@
+                                <Text style={pro_styles.text_style}>{this.state.verse}</Text>
+                            </View>
+                            {(this.state.interestEnabled || this.state.isEditing) &&
+                                <View>
+                                    <View style={{ flex: 1, flexDirection: 'row' }}>
+                                    <Text style={pro_styles.category_header}>Interests</Text>
+                                    {this.state.isEditing && <Switch
+                                        trackColor={{ false: "#CBCBCB", true: "#16B7EA" }}
+                                        onValueChange={this.toggleSwitchInterest}
+                                        value={this.state.editState.interestEnabled}
+                                        style={{ alignSelf: 'center', marginLeft: 'auto' }}
+                                    />}
+                                    </View>
+                                    <View style={{ flex: 1, flexDirection: 'row', flexWrap: 'wrap' }}>
+                                    {this.state.interests.map((interest) =>
+                                        <Interest interest={interest} key={interest} />
+                                    )}
+                                    </View>
+                                </View>
+                            }
+                            {(this.state.activityEnabled || this.state.isEditing) &&
+                                <View>
+                                    <View style={{ flex: 1, flexDirection: 'row' }}>
+                                    <Text style={pro_styles.category_header}>Recent activity</Text>
+                                    {this.state.isEditing &&
+                                        <Switch
+                                        trackColor={{ false: "#CBCBCB", true: "#16B7EA" }}
+                                        onValueChange={this.toggleSwitchActivity}
+                                        value={this.state.editState.activityEnabled}
+                                        style={{ alignSelf: 'center', marginLeft: 'auto' }}
+                                        />}
+                                    </View>
+                                    <View>
+                                    {this.state.activities.map((activity) =>
+                                        <Activity
+                                        question={activity.question}
+                                        answer={activity.answer}
+                                        key={activity.id}
+                                        />
+                                    )}
+                                    </View>
+                                </View>
+                            }
+                        </View>
+                    </View>
+                </ScrollView>
+            </SafeAreaView>
+        );
```

Comment:
Is there a way we could pull different components from here outwards so that this is easier to read and edit in the future? There's a lot here, which makes it difficult to parse. For example, maybe have a const ageInput set equal to its nested components, and call the ageInput inside the render function with {ageInput}, etc. for the other inputs and such.

Reply:
sure I can separate them into diff classes